src/analysis/retail/shaman/restoration/modules/talents/totemic/SurgingTotem.tsx

```diff
@@ -0,0 +1,391 @@
+      this._onRemoveBuff,
+    );
+    this.addEventListener(Events.fightend, this._onFightEnd);
+  }
+
+  _onFightEnd() {
+    this._rateCasts(this.SurgingTotemCasts);
+  }
+  get averageMaxTargets() {
+    return this.totalMaxTargets / this.casts;
+  }
+
+  get averageHitsPerTick() {
+    const totalHits = this.healingRainTicks.reduce((total, tick) => total + tick.hits, 0);
+    return totalHits / this.healingRainTicks.length;
+  }
+
+  suggestions(when: When) {
+    const suggestionThreshold = this.suggestionThreshold;
+    when(suggestionThreshold.actual)
+      .isLessThan(suggestionThreshold.isLessThan.minor)
+      .addSuggestion((suggest, actual, recommended) =>
+        suggest(
+          <span>
+            Try to always cast <SpellLink spell={SPELLS.HEALING_RAIN_TOTEMIC} /> in areas where
+            players stack. This allows the spell to consistently hit all possible targets. You can{' '}
+            {!this.selectedCombatant.hasTalent(TALENTS_SHAMAN.TOTEMIC_PROJECTION_TALENT) ? (
+              <>talent into </>
+            ) : (
+              <>use </>
+            )}
+            <SpellLink spell={TALENTS_SHAMAN.TOTEMIC_PROJECTION_TALENT} /> to reposition it every 10
+            seconds.
+          </span>,
+        )
+          .icon(SPELLS.HEALING_RAIN_TOTEMIC.icon)
+          .actual(
+            defineMessage({
+              id: 'shaman.restoration.suggestions.healingRain.averageTargets',
+              message: `${suggestionThreshold.actual.toFixed(2)} average targets healed`,
+            }),
+          )
+          .recommended(
+            `${suggestionThreshold.isLessThan.minor} average targets healed is recommended`,
+          )
+          .regular(suggestionThreshold.isLessThan.average)
+          .major(suggestionThreshold.isLessThan.average),
+      );
+  }
+
+  get suggestionThreshold() {
+    return {
+      actual: this.averageHitsPerTick,
+      isLessThan: {
+        minor: 4,
+        average: 3,
+        major: 2,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  onHealingRainHeal(event: HealEvent) {
+    // Filter out pets, but only if it fully overhealed as Rain will prioritize injured pets over non-injured players
+    // fully overhealing guarantees that there are not enough players in the healing rain
+    const combatant = this.combatants.getEntity(event);
+    if (!combatant && event.overheal && event.amount === 0) {
+      return;
+    }
+
+    const healingRainTick = this.healingRainTicks.find(
+      (tick) => event.timestamp - BUFFER_MS <= tick.timestamp,
+    );
+    if (!healingRainTick) {
+      this.healingRainTicks.push({
+        timestamp: event.timestamp,
+        hits: 1,
+      });
+    } else {
+      // dirty fix for partial ticks happening at the same time as a real tick
+      healingRainTick.hits =
+        healingRainTick.hits + 1 > this.maxTargets
+          ? (healingRainTick.hits = this.maxTargets)
+          : healingRainTick.hits + 1;
```

Comment:
don't need to do the 2nd assignment on line 165, right? this can just be:

```suggestion
      healingRainTick.hits = Math.min(this.maxTargets, healingRainTick.hits + 1);
```

Reply:
Correct. I will fix it as well in the original Healing rain module